src/main/java/appeng/items/storage/ItemReadOnlyStorageCell.java

```diff
@@ -0,0 +1,240 @@
+import appeng.api.exceptions.MissingDefinition;
+import appeng.api.implementations.items.IItemGroup;
+import appeng.api.implementations.items.IReadOnly;
+import appeng.api.implementations.items.IStorageCell;
+import appeng.api.implementations.items.IUpgradeModule;
+import appeng.api.storage.ICellInventory;
+import appeng.api.storage.ICellInventoryHandler;
+import appeng.api.storage.IMEInventoryHandler;
+import appeng.api.storage.StorageChannel;
+import appeng.api.storage.data.IAEItemStack;
+import appeng.api.storage.data.IItemList;
+import appeng.core.AEConfig;
+import appeng.core.features.AEFeature;
+import appeng.core.localization.GuiText;
+import appeng.items.AEBaseItem;
+import appeng.items.contents.CellConfig;
+import appeng.items.contents.CellUpgrades;
+import appeng.items.materials.MaterialType;
+import appeng.util.InventoryAdaptor;
+import appeng.util.Platform;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.util.List;
+import java.util.Set;
+
+
+public final class ItemReadOnlyStorageCell extends AEBaseItem implements IStorageCell, IItemGroup, IReadOnly
+{
+	private final int totalBytes;
+	private final int perType;
+	private final double idleDrain;
+
+	public ItemReadOnlyStorageCell()
+	{
+		this.setMaxStackSize( 1 );
+		this.totalBytes = 64 * 1024;
+		this.idleDrain = 2.0;
+		this.perType = 512;
+	}
+
+	@Override
+	public void addCheckedInformation( final ItemStack stack, final EntityPlayer player, final List<String> lines, final boolean displayMoreInfo )
+	{
+		final IMEInventoryHandler<?> inventory = AEApi.instance().registries().cell().getCellInventory( stack, null, StorageChannel.ITEMS );
+
+		if( inventory instanceof ICellInventoryHandler )
+		{
+			final ICellInventoryHandler handler = (ICellInventoryHandler) inventory;
+			final ICellInventory cellInventory = handler.getCellInv();
+
+			if( cellInventory != null )
+			{
+				if( getReadOnly( stack ) )
+				{
+					if( cellInventory.getUsedBytes() == 0 )
+					{
+						lines.add( GuiText.Empty.getLocal() );
+					}
+					else
+					{
+						lines.add( cellInventory.getUsedBytes() + " " + GuiText.BytesUsed.getLocal() );
+
+						lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Types.getLocal() );
+					}
+				}
+				else
+				{
+					lines.add( cellInventory.getUsedBytes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalBytes() + ' ' + GuiText.BytesUsed.getLocal() );
+
+					lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalItemTypes() + ' ' + GuiText.Types.getLocal() );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method appeng.items.storage.ItemReadOnlyStorageCell.addCheckedInformation(ItemStack, EntityPlayer, List, boolean) appears to call the same method on the same object redundantly [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)